echopype/echodata/sensor_ep_version_mapping/v05x_to_v06x.py

```diff
@@ -0,0 +1,1131 @@
+    The function directly modifies the input EchoData object.
+    """
+
+    # Map Beam --> Sonar/Beam_group1
+    if "Beam" in ed_obj.group_paths:
+        ed_obj._tree["Sonar"].add_child(ed_obj._tree["Beam"])
+        ed_obj._tree["Sonar/Beam"].name = "Beam_group1"
+
+    # Map Beam_power --> Sonar/Beam_group2
+    if "Beam_power" in ed_obj.group_paths:
+        ed_obj._tree["Sonar"].add_child(ed_obj._tree["Beam_power"])
+        ed_obj._tree["Sonar/Beam_power"].name = "Beam_group2"
+
+
+def get_channel_id(ed_obj, sensor):
+    """
+    Returns the channel_id for all non-unique frequencies.
+
+    Parameters
+    ----------
+    ed_obj : EchoData
+        EchoData object that was created using echopype version 0.5.x.
+    sensor : str
+        The sensor used to create the v0.5.x file.
+
+    Returns
+    -------
+    A datarray specifying the channel_ids with dimension frequency.
+    """
+    if sensor == "AZFP":
+
+        # create frequency_nominal variable
+        freq_nom = ed_obj["Sonar/Beam_group1"].frequency
+
+        # create unique channel_id for AZFP
+        freq_as_str = (freq_nom / 1000.0).astype(int).astype(str).values
+        channel_id_str = [
+            str(ed_obj["Sonar"].sonar_serial_number) + "-" + freq_as_str[i] + "-" + str(i + 1)
+            for i in range(len(freq_as_str))
+        ]
+        channel_id = xr.DataArray(
+            data=channel_id_str, dims=["frequency"], coords={"frequency": freq_nom}
+        )
+
+    else:
+
+        channel_id = xr.concat(
+            [child.ds.channel_id for child in ed_obj._tree["Sonar"].children], dim="frequency"
+        )
+
+    return channel_id
+
+
+def _frequency_to_channel(ed_obj, sensor):
+    """
+    1. In all groups that it appears, changes the dimension
+    ``frequency`` to ``channel`` whose values are based on
+    ``channel_id`` for EK60/EK80 and are a custom string
+    for AZFP.
+    2. Removes channel_id if it appears as a variable
+    3. Adds the variable ``frequency_nominal`` to all
+    Datasets that have dimension ``channel``
+
+    Parameters
+    ----------
+    ed_obj : EchoData
+        EchoData object that was created using echopype version 0.5.x.
+    sensor : str
+        The sensor used to create the v0.5.x file.
+
+    Notes
+    -----
+    The function directly modifies the input EchoData object.
+    """
+
+    channel_id = get_channel_id(ed_obj, sensor)  # all channel ids
+
+    for grp_path in ed_obj.group_paths:
+
+        if "frequency" in ed_obj[grp_path]:
+
+            # add frequency_nominal, remove standard_name attribute, and
+            # rename frequency to channel
+            ed_obj[grp_path]["frequency_nominal"] = ed_obj[grp_path].frequency
+
+            if "standard_name" in ed_obj[grp_path].frequency_nominal.attrs:
+                del ed_obj[grp_path].frequency_nominal.attrs["standard_name"]
+            ed_obj[grp_path] = ed_obj[grp_path].rename({"frequency": "channel"})
+
+            # set values for channel
```

Comment:
`ed_obj[grp_path]["channel"]` is float type, b/c it's simply the old `frequency` renamed. But here it's assigned string values. I'm either puzzled or pleasantly surprised that the assignment doesn't lead to an error. Is the `ed_obj[grp_path]["channel"]` data type being changed to string?

Reply:
Yes, I was very happy that xarray took these types of scenarios into consideration. If you are assigning a coordinate to a new array that is the same size, no error will occur even if they are of different types. I imagine that xarray points to positions in the coordinate array, rather than hard coding them in.